deploy/buildtools/buildconfig.py

```diff
@@ -2,176 +2,113 @@
 manager """
 
 from time import strftime, gmtime
-import configparser
 import pprint
+from importlib import import_module
 
-from runtools.runtime_config import RuntimeHWDB
 from awstools.awstools import *
+from buildtools.buildfarmhostdispatcher import *
+
+def inheritors(klass):
+    subclasses = set()
+    work = [klass]
+    while work:
+        parent = work.pop()
+        for child in parent.__subclasses__():
+            if child not in subclasses:
+                subclasses.add(child)
+                work.append(child)
+    return subclasses
 
 class BuildConfig:
-    """ This represents a SINGLE build configuration. """
-    def __init__(self, name, buildconfigdict, launch_time):
+    """ Represents a single build configuration used to build RTL/drivers/AFIs. """
+
+    def __init__(self, name, recipe_config_dict, build_farm_hosts_configfile, global_build_config, launch_time):
+        """ Initialization function.
+
+        Parameters:
+            name (str): Name of config i.e. name of build_recipe.yaml section
+            recipe_config_dict (dict): build_recipe.yaml options associated with name
+            build_farm_hosts_configfile (dict): Parsed representation of build_farm_hosts.yaml file
+            global_build_config (BuildConfigFile): Global build config file
+            launch_time (str): Time manager was launched
```

Comment:
I have not used any specific docstring format. I can do whatever as well (no preference).

Reply:
I think yours look close to what I know as Google style.  Let's use them and install Napoleon into our sphinx build so that they end up being formatted correctly in the docs build but I understand that it isn't happening in this PR.  I'll reformat the one I added to `awstools.launch_instances` or you can feel free to make them look more like yours the next time you merge from `main`.